pkg/kubectl/cmd/util/factory.go

```diff
@@ -199,17 +227,61 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clients: clients,
 		flags:   flags,
 
-		Object: func() (meta.RESTMapper, runtime.ObjectTyper) {
+		// If discoverDynamicAPIs is true, make API calls to the discovery service to find APIs that
+		// have been dynamically added to the apiserver
+		Object: func(discoverDynamicAPIs bool) (meta.RESTMapper, runtime.ObjectTyper) {
 			cfg, err := clientConfig.ClientConfig()
 			CheckErr(err)
 			cmdApiVersion := unversioned.GroupVersion{}
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
+			if discoverDynamicAPIs {
```

Comment:
cc @krousey I think you'd be interested in this. This is building REST mapper (with the exisiting RESTMapper struct) based on discovery results.